sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/implementation/util/BlobSasImplUtil.java

```diff
@@ -134,6 +137,12 @@ public BlobSasImplUtil(BlobServiceSasSignatureValues sasValues, String container
         this.contentType = sasValues.getContentType();
         this.authorizedAadObjectId = sasValues.getPreauthorizedAgentObjectId();
         this.correlationId = sasValues.getCorrelationId();
+        /*
+        Prefer the encryption scope explicitly set on the sas values. If none present, fallback to the value on the
+        client.
+         */
+        this.encryptionScope = sasValues.getEncryptionScope() == null
+            ? encryptionScope : sasValues.getEncryptionScope();
```

Comment:
few comments here.
-this doesn't seem to be right place for this logic - according to comment we're assuming who the caller is, so why can't we have this logic in the caller?
- from the pov of this class encryptionScope and sasValues.encrytptionscope are redundant.
- should we do such a silent fallback ? this might surprise some users. Consider throwing if encryption scope is specified in both places - this might help discover unintended usage.

Reply:
- I can move this logic to the caller
- I am also fine with throwing if it is present in both places

Reply:
Looking a little further into this, something we've done with snapshots is discourage people setting it on BlobServiceSasSignatureValues, telling them: "Please use the SAS generation methods provided on the desired (snapshot) blob client that will auto-populate the snapshot id." I think to be consistent with that, we should remove this setter from BlobServiceSasSignatureValues and just take the value from the property on the client

Reply:
not a bad idea.